### PR TITLE
Rlgr overrun

### PR DESCRIPTION
**Pull request by jsorg71**
Fix for a few issues with progressive remotefx.  Now it can handle when compressed buffer is too small.

### CODE REVIEW

File: src/rfxencode_dwt_shift_rem.c [hunk 1]
```diff
@@ -44,7 +44,7 @@
 
 #define SETUPLOQ(_index, _shift) do { \
     lo_fact = (((quants[_index] >> (_shift)) & 0xf) - 6) + DWT_FACTOR; \
-    lo_half = 1 << (hi_fact - 1); } while (0)
+    lo_half = 1 << (lo_fact - 1); } while (0)
```
Comment (Nexarian) [inline src/rfxencode_dwt_shift_rem.c:47]: I remember copying this code from your branch and being confused as to why it was hi_fact here :)

**Comment by Nexarian** [comment]
I don't fully understand what's going on here, but looks reasonable.

**Review by Nexarian — APPROVED** [review]

File: src/rfxencode_dwt_shift_rem.c (hunk 1 above)
Comment (jsorg71) [reply src/rfxencode_dwt_shift_rem.c:47]: oh, hehe, well your confusion was warranted.

**Comment by matt335672** [comment]
Same here!

**Pull request merged**